lib/prepare-sundials.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
```

Comment:
@wds15, could you document this script at a high-level? What would be extremely helpful:

1. Why / when a maintainer would use this.
2. What the maintainer needs to set up before using this.
3. What variables the maintainer needs to set in the script.
4. What the expected output is and how the maintainer is supposed to know this worked correctly.

Finally, we need to put this in the wiki. Probably here: [Updating Libraries](https://github.com/stan-dev/math/wiki/Updating-Libraries)

Reply:
What's `sundials-config`?

Reply:
sundials-config contains all stan specific configuration items for Sundials. I introduced it in my last commit. It serves to separate out things.